res/controllers/M-Audio-Xponent-Advanced-scripts.js

```diff
@@ -431,60 +473,57 @@ MaudioXponent.effectButton = function(channel, control, value, status, group) {
+            if (focusedEffect === button) {
+                // Toggle focused Effect
+                group = "[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "_Effect" + focusedEffect + "]";
+                var enabled = !engine.getValue(group, "enabled");
+                engine.setValue(group, "enabled", enabled);
+            } else {
+                // Focus effect
+                engine.setValue(group, "focused_effect", button);
+                engine.setValue(group, "show_focus", true);
+            }
+        }
     }
 };
 
-MaudioXponent.effectParameter = function(channel, control, value, status, group) {
-    // script.midiDebug(channel, control, value, status, group);
+MaudioXponent.effectKnob = function(channel, control, value) {
     var knob = control - 0x0B;
     var scaledValue = value / 0x7F;
 
-    if (MaudioXponent.state.focusedEffectUnit != 0) {
+    if (MaudioXponent.state.focusedEffectUnit === 0) {
         if (MaudioXponent.rightDeck.shift) {
-            // Manipulate focused EffectUnit parameters
-            if (knob == 4) {
-                // Wet / Dry
-                engine.setParameter("[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "]", "mix", scaledValue);
-            } else {
-                // Other parameter
-                engine.setParameter("[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "_Effect1]", "parameter" + knob, scaledValue);
-            }
+            // Manipulate Wet/Dry mix
+            engine.setParameter("[EffectRack1_EffectUnit" + knob + "]", "mix", scaledValue);
         } else {
-            // Manipulate focused EffectUnit meta/super knobs
-            if (knob == 4) {
-                // Super-knob
-                group = "[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "]";
-                engine.setParameter(group, "super1", scaledValue);
-            } else {
-                // Meta-knob
-                group = "[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "_Effect" + knob + "]";
-                engine.setParameter(group, "meta", scaledValue);
-            }
+            // Manipulate Unit Super-knobs
+            engine.setParameter("[EffectRack1_EffectUnit" + knob + "]", "super1", scaledValue);
         }
     } else {
-        if (MaudioXponent.rightDeck.shift) {
-            // Manipulate Wet/Dry mix
-            group = "[EffectRack1_EffectUnit" + knob + "]";
-            engine.setParameter(group, "mix", scaledValue);
+        // An effect has focus
+        var focusedEffect = engine.getValue("[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "]", "focused_effect");
+        
+        if (focusedEffect === 0) {
+            // Manipulate focused EffectUnit meta knobs
+            engine.setParameter("[EffectRack1_EffectUnit" + MaudioXponent.state.focusedEffectUnit + "_Effect" + knob + "]", "meta", scaledValue);
```

Comment:
> Can we say for sure that no skin will show more than three effects in a unit?

Yes, because this is what DJ controllers are designed for. I have [an idea](https://bugs.launchpad.net/mixxx/+bug/1656189) for how it may be practical to use arbitrarily long effects chains with the same 3 effect knobs + dry/wet, but don't count on that getting implemented any time soon.

> I'd personally love it if Mixxx gets detachable windows in the future. I live in a multi-monitor world all day long, and I can totally see dragging the library to a second screen which would leave more room on the primary monitor for things like effects. With more screen, you could have more effects showing.

That would be nice, and maybe it'll be possible when we move from the current skin system to QML. But the main use case for Mixxx is on a laptop. I don't know about you, but I don't want to drag a second screen to a gig.